air/src/proof/mod.rs

```diff
@@ -282,78 +309,116 @@ fn proven_security_protocol_for_m(
+    let trace_domain_size = trace_domain_size as f64;
+    let num_openings = 2.0;
+    let rho_plus = (trace_domain_size + num_openings) / lde_domain_size;
+    let alpha = (1.0 + rho_plus) * 0.5;
+    // we use the blowup factor in order to bound the max degree
+    let max_deg = options.blowup_factor() as f64 + 1.0;
 
-    // ALI related soundness error. Note that C here is equal to 1 because of our use of
-    // linear batching.
-    let ali_err_bits = -log2(l_plus) + extension_field_bits;
+    // we apply Theorem 3 in https://eprint.iacr.org/2024/1553
+    let mut epsilons_bits_neg = vec![];
+
+    // ALI related soundness error
+    let epsilon_1_bits_neg = extension_field_bits;
+    epsilons_bits_neg.push(epsilon_1_bits_neg);
 
     // DEEP related soundness error. Note that this uses that the denominator |F| - |D ∪ H|
     // can be approximated by |F| for all practical domain sizes. We also use the blow-up factor
-    // as an upper bound for the maximal constraint degree.
-    let deep_err_bits = -log2(
-        l_plus * (max_deg * (trace_domain_size + num_openings - 1.0) + (trace_domain_size - 1.0)),
-    ) + extension_field_bits;
+    // as an upper bound for the maximal constraint degree
+    let epsilon_2_bits_neg =
+        -log2(max_deg * (trace_domain_size + num_openings - 1.0) + (trace_domain_size - 1.0))
+            + extension_field_bits;
+    epsilons_bits_neg.push(epsilon_2_bits_neg);
+
+    // Computes FRI commit-phase (i.e., pre-query) soundness error
+    let epsilon_3_bits_neg = extension_field_bits - log2(lde_domain_size);
+    epsilons_bits_neg.push(epsilon_3_bits_neg);
+
+    // epsilon_i for i in [3..(k-1)], where k is number of rounds
+    let folding_factor = options.to_fri_options().folding_factor() as f64;
+    let num_fri_layers = options.to_fri_options().num_fri_layers(lde_domain_size as usize);
+    let epsilon_i_min_bits_neg = (0..num_fri_layers)
+        .map(|_| extension_field_bits - log2((folding_factor - 1.0) * (lde_domain_size + 1.0)))
+        .fold(f64::INFINITY, |a, b| a.min(b));
+    epsilons_bits_neg.push(epsilon_i_min_bits_neg);
 
-    let min = cmp::min(cmp::min(fri_err_bits, ali_err_bits as u64), deep_err_bits as u64);
-    if min < 1 {
-        return 0;
-    }
+    // Compute FRI query-phase soundness error
+    let epsilon_k_bits_neg = options.grinding_factor() as f64 - log2(powf(alpha, num_fri_queries));
+    epsilons_bits_neg.push(epsilon_k_bits_neg);
 
-    min - 1
+    // return the round-by-round (RbR) soundness error
+    epsilons_bits_neg.into_iter().fold(f64::INFINITY, |a, b| a.min(b)) as u64
 }
 
 // HELPER FUNCTIONS
 // ================================================================================================
 
-/// Computes the largest proximity parameter m needed for Theorem 8
-/// in <https://eprint.iacr.org/2022/1216.pdf> to work.
+/// Computes the largest proximity parameter m such that eta is greater than 0 in Theorem 1 in
+/// https://eprint.iacr.org/2021/582.
```

Comment:
Thm 1 in ethSTARK is the Johnson bound, which does not mention m at all. Maybe we should refer to the definition of eta here.

Reply:
Indeed, it is hidden in the proof. Expanded the comment in order to make it easier to get the point